### PR TITLE
CMake Updates to Replace Configuration Files

### PR DESCRIPTION
**Pull request by agustin-vaca**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -118,25 +127,54 @@ add_definitions (-DSG_DATA_TYPE=double)
...
+                
+                #Non-AVX platforms
+                elseif("${INTEL_PLATFORM}" STREQUAL "non_avx")
+                    #No specified flags
+                else()
+                    message("Platform was not recognized, no options passed.")
+                endif()
+            endif()
+
+            #Set compiler based on options
```
Comment (jyoung3131) [inline CMakeLists.txt:158]: This might be a separate option for MPICC as the compiler that supersedes the compiler?

**Comment by jyoung3131** [comment]
Let's comment out IBM XL compiler for now since we have limited test platforms for it.

**Comment by jyoung3131** [comment]
Add a small README named INSTALLATION.md or BUILD.md with options for input to CMake.

**Comment by jyoung3131** [comment]
@JDTruj2018 - can you check over this PR especially for your use case of CUDA + MPI?

We were not able to test IBM XL and Cray compilers, but these may not be widely used.

**Comment by jyoung3131** [comment]
The last commit with some sign changes was as a result of a set of warnings from NVHPC.

**Comment by JDTruj2018** [comment]
I am putting this here for reference (https://stackoverflow.com/questions/45933732/how-to-specify-a-compiler-in-cmake). I'll expand on it in a few of my suggested edits.

The main take-aways are:
1. We may want to avoid setting CMAKE_C_COMPILER, CMAKE_CXX_COMPILER, and CMAKE_CUDA_COMPILER in CMakelists.txt
2. If we do set CMAKE_C_COMPILER, CMAKE_CXX_COMPILER, and CMAKE_CUDA_COMPILER in CMakelists.txt, we need to do so before any project statements.

**Review by JDTruj2018 — COMMENTED** [review]
@agustin-vaca @jyoung3131 

Added 1 commit to fix errors with CUDA and CUDA+MPI on my end. 
Everything else looked good. I am okay with approving once tests pass.

**Review by JDTruj2018 — APPROVED** [review]

**Comment by JDTruj2018** [comment]
closes #132

**Review by JDTruj2018 — APPROVED** [review]

**Comment by jyoung3131** [comment]
Thanks, Jered! If I understand right from the link you shared, it seems like the ideal solution would be to push the compiler support into toolchain files in the future to avoid issues with default compiler flags. 

Since this is already a large change to the build process, I'm going to go ahead and approve it with the caveat that we should address the compiler-specific flags in a separate issue.

**Comment by JDTruj2018** [comment]
That sounds right. Essentially, we could leave it as the user responsibility to set `CMAKE_C_COMPILER`, etc. on the command line with `-DCMAKE_C_COMPILER`, and then generate the appropriate flags given that input.

The reason it came up is because with nvcc, CMake was trying to apply linkage flags that nvcc didn't recognize (pthreads with MPI for example, or the -fPIE flag) and failing. So ensuring the compiler CMake vars are set before defining a project or languages fixed the issue.

**Pull request merged**